src/lib/support/Pool.cpp

```diff
@@ -0,0 +1,72 @@
+/*
```

Comment:
This should also contain the original copyright since it's heavily leveraged from that base of code.

Reply:
done